In [80]:
import pandas as pd
import matplotlib.pyplot as plt
from random import randint
import numpy as np

In [13]:
#1. Create a Datatime index containing all the weekdays days of year 2019 and assign a random number to each of them in a dataframe. 
x = pd.Series([randint(0,10) for i in range(10)], index = pd.date_range('2019',  periods = 10))
print('output is:\n \n', x)

output is:
 
 2019-01-01    8
2019-01-02    9
2019-01-03    6
2019-01-04    8
2019-01-05    3
2019-01-06    3
2019-01-07    6
2019-01-08    7
2019-01-09    4
2019-01-10    2
Freq: D, dtype: int64


In [20]:
#2. Given Pandas series , height = [23,42,55] and weight = [71,32,48] . Create a dataframe with height and weight as column names. 
ht = pd.Series([23,42,55])
wt = pd.Series([71,32,48])
x = pd.DataFrame()
x['ht'] = ht
x['wt'] = wt
print('dataframe with height and weight as column names is: \n\n', x)

dataframe with height and weight as column names is: 

    ht  wt
0  23  71
1  42  32
2  55  48


In [26]:
#3. How to get the items of series A not present in series B .From ser1 remove items present in ser2.
ser1 = pd.Series([2,3,4,5])
ser2 = pd.Series([5,6,7,8])
x = pd.DataFrame()
y = pd.DataFrame()
x['ser1'] = ser1
x['ser2'] = ser2
y['ser1'] = ser1
y['ser2'] = ser2
y = y['ser1'][~ y.ser1.isin(y['ser2'])]
print('Items of Series A not present in Series B:\n\n', y)
x = x[~x.ser1.isin(x['ser2'])]
print('\nFrom Ser1 remove items present in Ser2: \n\n',x)

Items of Series A not present in Series B:

 0    2
1    3
2    4
Name: ser1, dtype: int64

From Ser1 remove items present in Ser2: 

    ser1  ser2
0     2     5
1     3     6
2     4     7


In [28]:
titanic = pd.read_csv(r'C:\Users\Insta\Desktop\End2End\endtoenddatasets\titanic.csv')

In [30]:
titanic.head()

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S

In [32]:
titanic.describe()

PassengerId    Survived      Pclass         Age       SibSp  \
count   891.000000  891.000000  891.000000  714.000000  891.000000   
mean    446.000000    0.383838    2.308642   29.699118    0.523008   
std     257.353842    0.486592    0.836071   14.526497    1.102743   
min       1.000000    0.000000    1.000000    0.420000    0.000000   
25%     223.500000    0.000000    2.000000   20.125000    0.000000   
50%     446.000000    0.000000    3.000000   28.000000    0.000000   
75%     668.500000    1.000000    3.000000   38.000000    1.000000   
max     891.000000    1.000000    3.000000   80.000000    8.000000   

            Parch        Fare  
count  891.000000  891.000000  
mean     0.381594   32.204208  
std      0.806057   49.693429  
min      0.000000    0.000000  
25%      0.000000    7.910400  
50%      0.000000   14.454200  
75%      0.000000   31.000000  
max      6.000000  512.329200

4. Compute the minimum, 25th percentile, median, 75th, and maximum of age in titanic dataset
from the above describe functions, we can say that
minimum age is 0.42 which is 42 months
25th percentile of age is 20.125 years - apprx 20 years
Median age is 28 years
75th percentile of age is 38 years
maximum age is 80 years

In [33]:
#5. How to get frequency counts of unique items of a series? Calculate the frequency counts of ‘SibSp’ column in titanic Dataset

titanic.SibSp.value_counts()

0    608
1    209
2     28
4     18
3     16
8      7
5      5
Name: SibSp, dtype: int64

In [34]:
#6. Keep only top 2 most frequent values as it is and replace everything else as ‘Other’ in ‘Embarked’ column of titanic dataset

titanic.Embarked.value_counts()

S    644
C    168
Q     77
Name: Embarked, dtype: int64

In [ ]:
7.  Bin the price column in titanic data set into 5 equal groups and get counts of each bin



In [41]:
#8. Count the number of missing values in each column?


print('\nCount of Missing Values in Titanic dataset:\n')
titanic.isnull().sum()


Count of Missing Values in Titanic dataset:



PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64

In [48]:
#9. Get the row number of the 5th largest value in the Age column of titanic dataset?
print('5th largest value in the age column of titanic dataset is:\n')
titanic.nlargest(5,['Age'])

5th largest value in the age column of titanic dataset is:



PassengerId  Survived  Pclass                                  Name  \
630          631         1       1  Barkworth, Mr. Algernon Henry Wilson   
851          852         0       3                   Svensson, Mr. Johan   
96            97         0       1             Goldschmidt, Mr. George B   
493          494         0       1               Artagaveytia, Mr. Ramon   
116          117         0       3                  Connors, Mr. Patrick   

      Sex   Age  SibSp  Parch    Ticket     Fare Cabin Embarked  
630  male  80.0      0      0     27042  30.0000   A23        S  
851  male  74.0      0      0    347060   7.7750   NaN        S  
96   male  71.0      0      0  PC 17754  34.6542    A5        C  
493  male  71.0      0      0  PC 17609  49.5042   NaN        C  
116  male  70.5      0      0    370369   7.7500   NaN        Q

we can say that fifth largest is 70.5 by using the above function nlarge

In [60]:
#11. Get the indices of items of ser2 in ser1 as a list.
ser1 = pd.Series([10,9,6,5,3,1,12,8,13])
ser2 = pd.Series([1,3,10,13])

print('indices of items of ser2 in ser 1:\n', ser1[ser1.isin(ser2)].index)

indices of items of ser2 in ser 1:
 Int64Index([0, 4, 5, 8], dtype='int64')


In [63]:
#12. How to convert a series of date-strings to a timeseries?

x = pd.Series(['01Jan2010','02-02-2011','20120303','2013/04/04','2014-05-05',
                        '2015-06-06T12:20'])
print('converted strings to timeseries:\n\n', pd.to_datetime(x))

converted strings to timeseries:

 0   2010-01-01 00:00:00
1   2011-02-02 00:00:00
2   2012-03-03 00:00:00
3   2013-04-04 00:00:00
4   2014-05-05 00:00:00
5   2015-06-06 12:20:00
dtype: datetime64[ns]


In [64]:
#13. Get the day of month, week number, day of year and day of week from ser.


In [77]:
#14. Compute the euclidean distance between series (points) p and q, without using a packaged formula.

p = pd.Series([1,2,3,4,5,6,7,8,9,10])
q = pd.Series([10,9,8,7,6,5,4,3,2,1])
x = sum((p - q)**2)**.5
print('Euclidian distance between p and q is: \n \n',
     round(x,3))

Euclidian distance between p and q is: 
 
 18.166


In [86]:
#15. How to create a TimeSeries starting ‘2000-01-01’ and 10 weekends (saturdays/sundays)
x = pd.Series(np.random.randint(1,10,10), pd.date_range('2000-01-01', periods=10, freq='W-SAT'))
print(' \nTimeSeries starting ‘2000-01-01’ and 10 weekends saturdays/sundays: \n\n', x)

 
TimeSeries starting ‘2000-01-01’ and 10 weekends saturdays/sundays: 

 2000-01-01    1
2000-01-08    6
2000-01-15    6
2000-01-22    5
2000-01-29    6
2000-02-05    4
2000-02-12    9
2000-02-19    6
2000-02-26    7
2000-03-04    6
Freq: W-SAT, dtype: int32


In [87]:
#16. Import every 50th row of BostonHousing dataset as a dataframe.

bostonhouse = pd.read_csv(r'C:\Users\Insta\Desktop\End2End\endtoenddatasets\boston_house_price.csv')
bostonhouse.head()

CRIM    ZN  INDUS  CHAS    NOX     RM   AGE     DIS  RAD    TAX  \
0  0.00632  18.0   2.31   0.0  0.538  6.575  65.2  4.0900  1.0  296.0   
1  0.02731   0.0   7.07   0.0  0.469  6.421  78.9  4.9671  2.0  242.0   
2  0.02729   0.0   7.07   0.0  0.469  7.185  61.1  4.9671  2.0  242.0   
3  0.03237   0.0   2.18   0.0  0.458  6.998  45.8  6.0622  3.0  222.0   
4  0.06905   0.0   2.18   0.0  0.458  7.147  54.2  6.0622  3.0  222.0   

   PTRATIO       B  LSTAT  MEDV  
0     15.3  396.90   4.98  24.0  
1     17.8  396.90   9.14  21.6  
2     17.8  392.83   4.03  34.7  
3     18.7  394.63   2.94  33.4  
4     18.7  396.90   5.33  36.2

In [91]:
bostonhouse1 = pd.DataFrame(bostonhouse.iloc[::50,:])
print('every 50th row of boston housing dataset is: \n',bostonhouse1)

every 50th row of boston housing dataset is: 
          CRIM    ZN  INDUS  CHAS    NOX     RM    AGE     DIS   RAD    TAX  \
0     0.00632  18.0   2.31   0.0  0.538  6.575   65.2  4.0900   1.0  296.0   
50    0.08873  21.0   5.64   0.0  0.439  5.963   45.7  6.8147   4.0  243.0   
100   0.14866   0.0   8.56   0.0  0.520  6.727   79.9  2.7778   5.0  384.0   
150   1.65660   0.0  19.58   0.0  0.871  6.122   97.3  1.6180   5.0  403.0   
200   0.01778  95.0   1.47   0.0  0.403  7.135   13.9  7.6534   3.0  402.0   
250   0.14030  22.0   5.86   0.0  0.431  6.487   13.0  7.3967   7.0  330.0   
300   0.04417  70.0   2.24   0.0  0.400  6.871   47.4  7.8278   5.0  358.0   
350   0.06211  40.0   1.25   0.0  0.429  6.490   44.4  8.7921   1.0  335.0   
400  25.04610   0.0  18.10   0.0  0.693  5.987  100.0  1.5888  24.0  666.0   
450   6.71772   0.0  18.10   0.0  0.713  6.749   92.6  2.3236  24.0  666.0   
500   0.22438   0.0   9.69   0.0  0.585  6.027   79.7  2.4982   6.0  391.0   

     PTRATIO    